arctic/store/version_store.py

```diff
@@ -576,6 +575,135 @@ def write(self, symbol, data, metadata=None, prune_previous_version=True, **kwar
+        Fast operation: Zero data/segment read/write operations.
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            dictionary of metadata to persist along with the symbol
+        prune_previous_version : `bool`
+            Removes previous (non-snapshotted) versions from the database.
+            Default: True
+        kwargs :
+            passed through to the write handler (only used if symbol does not already exist or is deleted)
+
+        Returns
+        -------
+        `VersionedItem`
+            VersionedItem named tuple containing the metadata of the written symbol's version document in the store.
+        """
+        # Make a normal write with empty data and supplied metadata if symbol does not exist
+        try:
+            previous_version = self._read_metadata(symbol)
+        except NoDataFoundException:
+            return self.write(symbol, data=None, metadata=metadata,
+                              prune_previous_version=prune_previous_version, **kwargs)
+
+        # Reaching here means that and/or metadata exist and we are set to update the metadata
+        new_version_num = self._version_nums.find_one_and_update({'symbol': symbol},
+                                                            {'$inc': {'version': 1}},
+                                                            upsert=True, new=True)['version']
+
+        # Populate the new version entry, preserving existing data, and updating with the supplied metadata
+        version = {k: previous_version[k] for k in previous_version.keys()}
+        version['_id'] = bson.ObjectId()
+        version['version'] = new_version_num
+        version['metadata'] = metadata
+        version['base_version_id'] = previous_version.get('base_version_id', previous_version['_id'])
+
+        return self._add_new_version_using_reference(symbol, version, previous_version, prune_previous_version)
+
+    def restore_version(self, symbol, as_of, prune_previous_version=True, allow_secondary=None):
+        """
+        Restore the specified 'symbol' data and metadata to the state of a given version/snapshot/date.
+        Returns a VersionedItem object only with a metadata element.
+        Fast operation: Zero data/segment read/write operations.
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        as_of : `str` or `int` or `datetime.datetime`
+            Return the data as it was as_of the point in time.
+            `int` : specific version number
+            `str` : snapshot name which contains the version
+            `datetime.datetime` : the version of the data that existed as_of the requested point in time
+        prune_previous_version : `bool`
+            Removes previous (non-snapshotted) versions from the database.
+            Default: True
+        allow_secondary : `bool` or `None`
```

Comment:
I don't think `allow_secondary` should be on this API. It's only there for scaling out reads

Reply:
removed